File: TestData/test_server.rb
require 'output'
require 'scriptbase'

class MainClass < ScriptBase

	attr_accessor :additional_string

	def on_receive (msg, outputObject, source)

		outputObject.add_metadata "TestHeader", "baz"

		OP::msg "GINININININ"
		if msg.payload.has "text"

			OP::msg "GINININININ"
			outputObject.payload = msg.payload.text

			if @additional_string != nil

				outputObject.payload = outputObject.payload + @additional_string

			end
		
		elsif (msg.payload.has("ob") && msg.payload.ob.has("bar"))
			
			outputObject.payload = { "foo" => (msg.payload.ob.bar * 10), "bar" => "baz" } 
		
		else

			outputObject.payload = "ARGH!½!!"
 
		end

		OP::msg "ARGH"
		return outputObject

	end

end

self.main_class = MainClass.new()

File: TestData/test_server.py

class MainClass:

	def r2_init(self):
		self.additional_string = None

	def on_receive(self, msg, outputObject, source):
		outputObject.AddMetadata("TestHeader", "baz")

		if msg.Payload.Has("text"):

			outputObject.Payload = msg.Payload.text

			if self.additional_string is not None:
				outputObject.Payload = outputObject.Payload + self.additional_string

		elif msg.Payload.Has("ob") and msg.Payload.ob.Has("bar"):
			outputObject.Payload = {}
			outputObject.Payload["foo"] = msg.Payload.ob.bar * 10
			outputObject.Payload["bar"] = "baz"
		else:
			outputObject.Payload = "ARGH!!!"

		return outputObject

main_class = MainClass()